apps/myjobs/app/controllers/workflows_controller.rb

```diff
@@ -13,7 +13,7 @@ def index
 
     @selected_id = session[:selected_id]
     session[:selected_id] = nil
-    @workflows = Workflow.preload(:jobs)
+    @workflows = Workflow.all
```

Comment:
Problem here is then we introduce an n+1 query problem. 1 query to get all the workflow objects, and then when we iterate over each workflow object to get job status information, we execute one query for each job.

The ideal solution would be to implement pagination, when we are only every querying a subset, and then to preserve the job records of each job submitted.

The current implementation enforces a 1:1 workflow to job record.  In this case though, we only have a 1:1 workflow+job and we might be able to get by using `Workflow.eager_load(:jobs)` instead.

So we need to investigate another solution to just dropping preload.

Reply:
I'm able to confirm that `Workflow.eager_load(:jobs)` works as well. It loads all 1000 records and their associated job if it exists.